modules/integration/src/test/java/org/wso2/messaging/integration/QueueConsumerTest.java

```diff
@@ -194,14 +194,15 @@ public void testConsumerWithBasicRecover(String port,
 
         subscriberSession.recover();
 
+        // Message order flip after recovering
         message = consumer.receive(5000);
-        Assert.assertNotNull(message, "Requeued first Message was not received");
-        Assert.assertTrue(message.getJMSRedelivered(), "Redelivered flag was not set in first message");
+        Assert.assertNotNull(message, "Requeued second message was not received");
+        Assert.assertFalse(message.getJMSRedelivered(), "Redelivered flag was set in second message" + message);
```

Comment:
Shall we use correlation id to track the messages rather than the execution order?